types/cli-progress/index.d.ts

```diff
@@ -144,48 +154,74 @@ export interface Preset {
     format: string;
 }
 
-export class SingleBar extends EventEmitter {
+export class GenericBar extends EventEmitter {
     /** Initialize a new Progress bar. An instance can be used multiple times! it's not required to re-create it! */
     constructor(opt: Options, preset?: Preset);
 
-    calculateETA(): void;
-    /** Force eta calculation update (long running processes) without altering the progress values. */
-    updateETA(): void;
+    /** Internal render function */
+    render(forceRendering?: boolean): void;
 
-    formatTime(t: any, roundToMultipleOf: any): any;
+    /** Starts the progress bar and set the total and initial value */
+    start(total: number, startValue: number, payload?: object): void;
 
-    getTotal(): any;
+    /** Stops the progress bar and go to next line */
+    stop(): void;
+
+    /** Sets the current progress value and optionally the payload with values of custom tokens as a second parameter */
+    update(current: number, payload?: object): void;
+    update(payload: object): void;
+
+    /** Calculate the actual progress value */
+    getProgress(): number;
 
     /** Increases the current progress value by a specified amount (default +1). Update payload optionally */
     increment(step?: number, payload?: object): void;
     increment(payload: object): void;
 
-    render(): void;
+    /** Get the total (limit) value */
+    getTotal(): number;
 
     /** Sets the total progress value while progressbar is active. Especially useful handling dynamic tasks. */
     setTotal(total: number): void;
 
-    /** Starts the progress bar and set the total and initial value */
-    start(total: number, startValue: number, payload?: object): void;
+    /** Force eta calculation update (long running processes) without altering the progress values. */
+    updateETA(): void;
+}
 
-    /** Stops the progress bar and go to next line */
-    stop(): void;
+export class SingleBar extends GenericBar {
+    /** Initialize a new Progress bar. An instance can be used multiple times! it's not required to re-create it! */
+    constructor(opt: Options, preset?: Preset);
 
-    stopTimer(): void;
+    /** Internal render function */
+    render(): void;
 
     /** Sets the current progress value and optionally the payload with values of custom tokens as a second parameter */
     update(current: number, payload?: object): void;
     update(payload: object): void;
+
+    /** Starts the progress bar and set the total and initial value */
+    start(total: number, startValue: number, payload?: object): void;
+
+    /** Stops the progress bar and go to next line */
+    stop(): void;
 }
 
 export class MultiBar extends EventEmitter {
     constructor(opt: Options, preset?: Preset);
 
-    create(total: number, startValue: number, payload?: any): SingleBar;
+    /** add a new bar to the stack */
+    create(total: number, startValue: number, payload?: any, barOptions?: object): SingleBar;
```

Comment:
isn't that options same object used to create an instance of a bar via constructor? (so not a generic object, but `Options`)

Reply:
tbh I was wondering about that. `barOptions` is used in one place (Object.assign) and seems to swallow everything. I'll change it to Options as this seems to be the intention.